paimon-core/src/main/java/org/apache/paimon/table/source/DataSplit.java

```diff
@@ -89,6 +92,16 @@ public long rowCount() {
         return rowCount;
     }
 
+    @Override
+    public boolean needsMerging() {
+        return rawFiles.isEmpty();
+    }
+
+    @Override
+    public List<RawFile> rawFiles() {
```

Comment:
We have two choices:
1. `Optional<List<RawFile>> convertToRawFiles`.
2. `boolean needsMerging()` and `List<RawFile> rawFiles()`, if need to merge, rawFiles should throws exception.

Maybe choice first is better.